x-pack/plugins/security_solution/server/endpoint/services/actions/utils.ts

```diff
@@ -116,37 +113,58 @@ export const getActionCompletionInfo = (
     completedAt: undefined,
     errors: undefined,
     outputs: {},
+    agentState: {},
     isCompleted: Boolean(agentIds.length),
     wasSuccessful: Boolean(agentIds.length),
   };
 
   const responsesByAgentId: ActionResponseByAgentId = mapActionResponsesByAgentId(actionResponses);
 
   for (const agentId of agentIds) {
-    if (!responsesByAgentId[agentId] || !responsesByAgentId[agentId].isCompleted) {
+    const agentResponses = responsesByAgentId[agentId];
+
+    // Set the overall Action to not completed if at least
+    // one of the agent responses is not complete yet.
+    if (!agentResponses || !agentResponses.isCompleted) {
```

Comment:
Nice! So on agent details and responder logs we use the agent-specific action state but on the page level (that I'm working on) we use this top-level status. 👍